apps/emqx/src/emqx_keepalive.erl

```diff
@@ -103,6 +95,10 @@ check(
 %% typically this is a few minutes.
 %% The maximum value is (65535s) 18 hours 12 minutes and 15 seconds.
 %% @doc Update keepalive's interval
--spec set(interval, non_neg_integer(), keepalive()) -> keepalive().
-set(interval, Interval, KeepAlive) when Interval >= 0 andalso Interval =< 65535000 ->
-    KeepAlive#keepalive{interval = Interval}.
+-spec set(interval, non_neg_integer(), non_neg_integer(), keepalive()) -> keepalive().
+set(interval, StatVal, Interval, undefined) when ?IS_INTERVAL(Interval) ->
+    init(StatVal, Interval);
+set(interval, StatVal, Interval, #keepalive{}) when ?IS_INTERVAL(Interval) ->
+    init(StatVal, Interval);
+set(interval, StatVal, Interval, _KeepAlive) ->
+    throw({bad_keepalive_interval, #{statval => StatVal, interval => Interval}}).
```

Comment:
this is not covered in test. is it even possible to happen?

Reply:
done